graph/cytoscape/cytoscape.go

```diff
@@ -139,7 +140,7 @@ func walk(sn *tree.ServiceNode, ndParent *NodeData, nodes *[]*NodeWrapper, edges
 	nd, found := findNode(nodes, name, sn.Version)
 
 	if !found {
-		nodeId := fmt.Sprintf("n%v", *nodeIdSequence)
+		nodeId := uuid.Must(uuid.NewV4()).String()
```

Comment:
I think we need to provide the same "id" for the same element if the graph didn't changed, but not sure.
@jshaughn do you know?

Reply:
I'm not sure about it, we may need to ask @jmazzitelli.  If we need consistent ids maybe we should hash the servicename+serviceversion.  Mazz did the recent work on updating the existing graph and it may require the id remain the same.  Even today the id is not guaranteed to stay the same, although typically it will, maybe a hash is a better approach.

Reply:
Right now, the service graph does not do any kind of "diff-ing" to remove obsolete nodes, add new nodes, or update existing nodes. So right now we do not need any kind of ID that needs to stay the same between different iterations of the graph. All we do now in the code is take the newest graph JSON and overwrite the entire existing graph nodes with the new nodes - basically we replace the entire old graph with an entire new graph - the old stuff is removed, and new stuff added - see : https://github.com/kiali/kiali-ui/blob/0d3d59de84d6e8182b837c2d47e18de8863f648c/src/components/CytoscapeGraph/CytoscapeGraph.tsx#L202-L203

That said, if we ever DO want to change this in the future (perhaps for performance reasons?) and we DO want to diff the old and new graph and only update the graph with new/removed/updated nodes and edges, then we WILL need some kind of UUID that lives across different graph iterations (so if I call the server-side and get the graph JSON at 1pm, and then I make another request at 2pm to get the JSON graph data, I will want to be able to know what nodes and edges were added/removed/updated compared to the 1pm graph data I requested earlier.).

So we don't need UUIDs now, but if we don't do it now, we may limit ourselves in the future. If it is easy to add a UUID to the data now, I say do it now. It may be useful/necessary in the future.

Reply:
+1 for servicename+serviceversion hash or namespace+servicename+serviceversion

Reply:
@jmazzitelli checking the doc on `json` it hints that it does some kind of diff-ing
> When setting cy.json({ elements: ... })
>
> the included elements are mutated as specified (i.e. as they would be by ele.json( eleJson )),
> the included elements not in the graph are added, and
> the not included elements are removed from the graph.

So, elements already there are mutated (and not created)